### PR TITLE
feat(curriculum): create tooling and deployment quiz

### LINKED ISSUE #56238: [HACKTOBERFEST 2024] - Tooling and Deployment Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Tooling and Deployment quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment`.

**All quiz questions need to relate to one of these topics:**

- Common bundlers (rollup, parcel, vite)
- Configure Github tooling to create preview deployments
- Custom domains, serverless functions, form submissions
- access control and integration with identity providers like Auth0
- Common deployment options (Netlify, Vercel, VPS)

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by SelormDev**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56238

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md [hunk 1]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+An Entry File
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+GitHub Actions offers features that let you _______
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md:130]: Not sure about this one. Technically all of the answers, except maybe user access control, would fit.

File: curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md [hunk 2]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+A `.github/workflows` YAML file
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+These are examples of events to trigger deployment workflow except ______
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md:218]:
```suggestion
Which of these is not an event that can trigger a GitHub Action?
```

File: curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md [hunk 3]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`HTML`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the function of the post method in a form?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md:284]:
```suggestion
What is the function of the `POST` method in a form?
```

File: curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md [hunk 4]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+To authenticate users
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Why is it important to integrate an identity provider in mordern application?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md:328]:
```suggestion
Why is it important to integrate an identity provider in a modern application?
```

File: curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md [hunk 5]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+GitHub Pages
 
 #### --answer--
 
-Placeholder answer
-
+VPS
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md:454]:
```suggestion
VPS

```

Restore the EOF new line.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md (hunk 5 above)
Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md:454]: resolving the conversation doesn't mean that the issue is fixed, please actually do the thing @SelormDev

Comment (SelormDev) [reply curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md:454]: Pardon me but I thought commiting the suggestion will apply the changes requested. Let me manually check and do the necessary changes.

Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md:454]: the diff still shows a removed line

image

Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md:454]: now it's fine, I committed the suggestion

**Review by naomi-lgbt — APPROVED** [review]

**Review by gikf — APPROVED** [review]

**Comment by gikf** [comment]
Hi @SelormDev. Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**